doc/extdev/index.rst

```diff
@@ -22,6 +22,34 @@ The configuration file itself can be treated as an extension if it contains a
 ``setup()`` function.  All other extensions to load must be listed in the
 :confval:`extensions` configuration value.
 
+Discovery of builders by entry point
+------------------------------------
+
+.. versionadded:: 1.6
+
+:term:`Builder` extensions can be discovered by means of `entry points`_ so
+they do not have to be listed in the :confval:`extensions` configuration value.
+
+Builder extensions need to define an entry point in the ``sphinx.builders``
+group. The name of the entry point needs to be set to the name of the builder,
+which is the name passed to the :option:`sphinx-build -b` option. Here is an
+example of how an entry point for 'mybuilder' can be defined in ``setup.py``::
+
+    setup(
+        # ...
+        entry_points={
+            'sphinx.builders': [
```

Comment:
I don't know if there are any sort of official guidelines for entry point group names, but the [setuptools documentation](https://setuptools.readthedocs.io/en/latest/setuptools.html#entry-points) seems to suggest the `sphinx.builders` format. But to remain consistent, you might want to opt for `sphinx_builders`.

The class **is** use; to instantiate the builder which is returned from `create_builder()`. An entry point needs to point to a symbol. Perhaps its possible to point it to a module, but that seems awkward.